cmp/tab/TabContainerModel.js

```diff
@@ -134,6 +122,57 @@ export class TabContainerModel {
         }
     }
 
+    //-----------------------------
+    // Tab addition, modification
+    //-----------------------------
+    /**
+     * Set the Tabs displayed by this object.
+     *
+     * @param {Object[]|TabModel[]} tabs - TabModels or configs for TabModels.
+     */
+    @action
+    setTabs(tabs) {
+        const oldTabs = this.tabs,
+            isInit = (oldTabs === null);
+        throwIf(!isInit && this.route, 'Dynamic tabs not available on TabContainer with routing.');
+        throwIf(!isInit && XH.isMobileApp, 'Dynamic tabs not available on mobile TabContainer.');
+
+        ensureUniqueBy(tabs, 'id', 'Multiple tabs have the same id.');
+
+        tabs = tabs.filter(p => !p.omit);
+        this.activeTabId = this.initialActiveTabId(tabs);
+        const newTabs = tabs.map(t => (
+            t instanceof TabModel ? t : new TabModel({...t, containerModel: this})
+        ));
+
+        this.tabs = newTabs;
+        if (oldTabs) {
+            XH.safeDestroy(difference(oldTabs, newTabs));
+        }
+    }
+
+    /**
+     * Add a Tab for display.
+     * @param {Object} tab - TabModel or config for TabModel to be added.
+     * @param {number} [index] - index in tab collection where tab is to be added
+     */
+    addTab(tab, index = this.tabs.length) {
+        const {tabs} = this;
+        this.setTabs([...tabs.slice(0, index), tab, ...tabs.slice(index)]);
+    }
+
+    /**
+     * Remove a Tab for display.
+     * @param {Object|String} tab - TabModel or id of TabModel to be removed.
+     */
+    removeTab(tab) {
```

Comment:
Hi! I can't leave a comment on the non-diff, but in line 206, we should not activate a tab if it's not found. (if you were to add an onClick 'x' handler to the tab title, it would try to activate what's been removed)

Reply:
Fixed!